client/src/redux/failed-updates-epic.js

```diff
@@ -45,7 +50,15 @@ function failedUpdateEpic(action$, state$) {
     filter(() => store.get(key)),
     filter(() => isOnlineSelector(state$.value)),
     tap(() => {
-      const failures = store.get(key) || [];
+      let failures = store.get(key) || [];
+
+      // delete unsubmitable failed challenges
+      if (!failures.every(isSubmitable)) {
```

Comment:
I could be wrong but I believe the extra iteration through the failures array (with the `every` method`) is not needed.
```suggestion
      let submitableFailures = failures.filter(isSubmitable);
      if (submitableFailures.length !== failures.length) {
        failures = submitableFailures;
        store.set(key, submitableFailures);
     }
```

Reply:
That is correct. the second iteration is not needed.